boot/seal.go

```diff
@@ -48,23 +48,23 @@ var (
 )
 
 // Hook functions setup by devicestate to support device-specific full
-// disk encryption implementations.
+// disk encryption implementations. The state much be locked when these
+// functions are called.
 var (
 	HasFDESetupHook = func() (bool, error) {
 		return false, nil
 	}
-	RunFDESetupHook = func(op string, params *FdeSetupHookParams) error {
-		return fmt.Errorf("internal error: RunFDESetupHook not set yet")
+	RunFDESetupHook = func(op string, params *FDESetupHookParams) ([]byte, error) {
+		return nil, fmt.Errorf("internal error: RunFDESetupHook not set yet")
 	}
 )
 
-// FdeSetupHookParams contains the inputs for the fde-setup hook
-type FdeSetupHookParams struct {
+// FDESetupHookParams contains the inputs for the fde-setup hook
+type FDESetupHookParams struct {
 	Key     secboot.EncryptionKey
 	KeyName string
 
-	KernelInfo *snap.Info
-	Model      *asserts.Model
+	Model *asserts.Model
```

Comment:
we should remember that if we have something like reseal (and if the implementations do take into account even just model we will have to), here we might need to pass a list of models, not just one

Reply:
I updated the code to pass a list of models and also pass this list to the helpers. This requires a change in the fde-setup protocol so Robert/Ondra who work on this need to be informed but AIUI they don't use the model today so that should be fine.